api/app/controllers/mno_enterprise/jpi/v1/admin/organizations_controller.rb

```diff
@@ -117,6 +122,78 @@ def invite_member
       @user = user.confirmed? ? invite : user
     end
 
+    def download_batch_example
+      path = File.join(File.dirname(File.expand_path(__FILE__)), '../../../../../assets/batch-example.csv')
+      send_file(path, filename: 'batch-example.csv', type: 'application/csv')
+    end
+
+    # POST /mnoe/jpi/v1/admin/organization/batch_import
+    # Organizations and Users batch import endpoint
+    # The file will be entirely rejected if one line has a validation failure
+    # Each line can be used to create a new customer organisation or a new customer user.
+    # Please use an external reference (external_id) to identify a company if you wish to add multiple users to the same company
+    # the import tool will not identify organisation based on names and will create a new organisation if no external reference is used.
+    # When an external reference if found, the organisation record is updated.
+    # If user email address already exists, it will be linked to the customer
+    def batch_import
+      file = params[:file]
+      # get the file's temporal path
+      file_temp_path = file.tempfile.path
+      # Read CSV File
+      begin
+        csv = CSV.read(file_temp_path, CSV_OPTIONS)
+      rescue CSV::MalformedCSVError => e
+        return render json: ["Could not Process CSV File: #{e.message}"], status: :bad_request
+      end
+      errors = validate_csv(csv)
+      if errors.any?
+        return render json: errors, status: :bad_request
+      end
+      @added_organizations = []
+      @updated_organizations = []
+      @added_users = []
+      @updated_users = []
+      # TODO Move to a Job
+      csv.each do |row|
```

Comment:
`each_with_index do |row, index|`

Reply:
Unfortunately CSV does not implement that method.